src/Components/RingdownCard.js

```diff
@@ -8,37 +8,102 @@ import Drawer from './Drawer';
 import Ringdown from '../Models/Ringdown';
 import RingdownDetails from './RingdownDetails';
 import RingdownEta from './RingdownEta';
+import Alert from './Alert';
 import './RingdownCard.scss';
 
-function RingdownCard({ className, ringdown }) {
+function RingdownCard({ className, ringdown, onStatusChange }) {
   const [isExpanded, setExpanded] = useState(false);
+  const [showCancel, setShowCancel] = useState(false);
+  const [showRedirect, setShowRedirect] = useState(false);
+  const [currentHospital] = useState(ringdown.hospital.id);
+
+  function handleCancel() {
+    setShowCancel(false);
+    onStatusChange(ringdown, Ringdown.Status.CANCEL_ACKNOWLEDGED);
+  }
+
+  function handleRedirect() {
+    setShowRedirect(false);
+    onStatusChange(ringdown, Ringdown.Status.REDIRECT_ACKNOWLEDGED);
+  }
 
   return (
     <Card
       className={classNames(
         'ringdown-card',
         { 'ringdown-card--offloaded': ringdown.currentDeliveryStatus === Ringdown.Status.OFFLOADED },
+        { 'ringdown-card--cancelled': ringdown.currentDeliveryStatus === Ringdown.Status.CANCELLED },
+        {
+          'ringdown-card--cancelled':
+            ringdown.currentDeliveryStatus === Ringdown.Status.REDIRECTED &&
+            ringdown.hospital.id &&
+            ringdown.hospital.id === currentHospital,
```

Comment:
I'm not sure I understand yet why these hospital id checks are needed... when an EMS user redirects a ringdown, the existing ringdown is marked REDIRECTED, and then the ringdown is cloned into a new object before changing the hospital id. If we're seeing the new cloned ringdown in REDIRECTED state in the EMS interface, then I think that means the hospital status is not properly being cleared in the clone. Check Ringdown.js clone() method, and try seeing if we can get rid of all these additional conditional checks.